src/coord/rmr/chan.c

```diff
@@ -88,6 +88,7 @@ void *MRChannel_Pop(MRChannel *chan) {
   pthread_mutex_lock(&chan->lock);
   while (!chan->size) {
     if (!chan->wait) {
+      chan->wait = true;  // reset the flag
```

Comment:
Since we now necessarily only unblock a single reader, should we change the broadcast call to a signal in the unblock function?

Reply:
what are the risks? what is the benefit?

Reply:
future-proofing, if we ever want to have multiple readers. We now unblock a single reader, so why wake everyone up?

Reply:
ok